glyphs-reader/src/font.rs

```diff
@@ -264,3 +294,296 @@ impl Path {
         self.nodes.reverse();
     }
 }
+
+fn for_raw_glyphs(
+    glyphs_file: &path::Path,
+    root_dict: &mut BTreeMap<String, Plist>,
+    callback: fn(&mut BTreeMap<String, Plist>) -> Result<(), Error>,
+) -> Result<(), Error> {
+    if !root_dict.contains_key("glyphs") {
+        return Ok(());
+    }
+    let Plist::Array(glyphs) = root_dict.get_mut("glyphs").unwrap() else {
+        return Err(Error::ParseError(glyphs_file.to_path_buf(), "Must have a glyphs array".to_string()));
+    };
+    for glyph in glyphs.iter_mut() {
+        let Plist::Dictionary(glyph) = glyph else {
+            return Err(Error::ParseError(glyphs_file.to_path_buf(), "Glyph must be a dict".to_string()));
+        };
+        callback(glyph)?;
+    }
+    Ok(())
+}
+
+fn fix_glyphs_named_infinity(
+    glyphs_file: &path::Path,
+    root_dict: &mut BTreeMap<String, Plist>,
+) -> Result<(), Error> {
+    for_raw_glyphs(glyphs_file, root_dict, |glyph| {
+        if !glyph.contains_key("glyphname") {
+            return Ok(());
+        }
+        if let Plist::Float(..) = glyph.get("glyphname").unwrap() {
+            glyph.insert(
+                "glyphname".to_string(),
+                Plist::String("infinity".to_string()),
+            );
+        }
+        Ok(())
+    })
+}
+
+fn weight_name(weight: i64) -> &'static str {
+    match weight {
+        100 => "Thing",
+        200 => "Extra-light",
+        300 => "Light",
+        400 => "Regular",
+        500 => "Medium",
+        600 => "Semi-bold",
+        700 => "Bold",
+        800 => "Extra-bold",
+        900 => "Black",
+        _ => panic!("No name for weight {}", weight),
+    }
+}
+
+impl Font {
+    fn is_v2(&self) -> bool {
+        let mut is_v2 = true;
+        if let Some(Plist::Integer(version)) = self.other_stuff.get(".formatVersion") {
+            is_v2 = *version < 3; // .formatVersion is only present for v3+
+        }
+        is_v2
+    }
+
+    fn v2_to_v3_axes(&mut self) -> Result<Vec<String>, Error> {
+        let mut tags = Vec::new();
+        if let Some(Plist::Array(custom_params)) = self.other_stuff.get_mut("customParameters") {
```

Comment:
you want to get the list of axes defined in the font. As usual, it's a bit more complicated than simply checking if an "Axes" custom parameter is defined :)

That parameter is optional and can be omitted, in which case one has to assume the predefined triplet of default axes definitions, i.e. weight, width and "custom", any of which may be used by a given font master.

see `get_axis_definitions` method from glyphsLib (this is used to define the designspace <axes> element):
 https://github.com/googlefonts/glyphsLib/blob/f6e9c4a29ce764d34c309caef5118c48c156be36/Lib/glyphsLib/builder/axes.py#L530

If you like, since you're only supporting one-axis wght fonts for the time being, you could assume the absence of "Axes" custom param means an implicit {"name": "Axes", "value": [{"Name": "Weight", "Tag": "wght"}]}

Reply:
Added a test and fixed, ty.